import json
import sys
import os
import subprocess
import shlex
import entrezpy.esearch.esearcher
#import entrezpy.esearch.esearcher.Esearcher

# {'activity_status': 'active', 'browsers': [{'label': 'BioPortal', 'title': 'BioPortal Browser', 'url': 'http://bioportal.bioontology.org/ontologies/BFO?p=classes'}], 'contact': {'email': 'phismith@buffalo.edu', 'github': 'phismith', 'label': 'Barry Smith'}, 'depicted_by': 'https://avatars2.githubusercontent.com/u/12972134?v=3&s=200', 'description': 'The upper level ontology upon which OBO Foundry ontologies are built.', 'domain': 'upper', 'homepage': 'http://ifomis.org/bfo/', 'id': 'bfo', 'in_foundry_order': 1, 'layout': 'ontology_detail', 'license': {'label': 'CC-BY', 'logo': 'http://mirrors.creativecommons.org/presskit/buttons/80x15/png/by.png', 'url': 'http://creativecommons.org/licenses/by/4.0/'}, 'mailing_list': 'https://groups.google.com/forum/#!forum/bfo-discuss', 'ontology_purl': 'http://purl.obolibrary.org/obo/bfo.owl', 'products': [{'id': 'bfo.owl', 'ontology_purl': 'http://purl.obolibrary.org/obo/bfo.owl'}, {'id': 'bfo.obo', 'ontology_purl': 'http://purl.obolibrary.org/obo/bfo.obo'}], 'title': 'Basic Formal Ontology', 'tracker': 'https://github.com/BFO-ontology/BFO/issues', 'usages': [{'description': 'BFO is imported by multiple OBO ontologies to standardize upper level structure', 'type': 'owl_import', 'user': 'http://obofoundry.org'}]}
# {'activity_status': 'active', 'alternatePrefix': 'ChEBI', 'browsers': [{'label': 'CHEBI', 'title': 'EBI CHEBI Browser', 'url': 'http://www.ebi.ac.uk/chebi/chebiOntology.do?treeView=true&chebiId=CHEBI:24431#graphView'}], 'build': {'infallible': 1, 'method': 'obo2owl', 'source_url': 'ftp://ftp.ebi.ac.uk/pub/databases/chebi/ontology/chebi.obo'}, 'contact': {'email': 'gowen@ebi.ac.uk', 'github': 'G-Owen', 'label': 'Gareth Owen'}, 'description': "A structured classification of molecular entities of biological interest focusing on 'small' chemical compounds.", 'domain': 'biochemistry', 'homepage': 'http://www.ebi.ac.uk/chebi', 'id': 'chebi', 'in_foundry_order': 1, 'layout': 'ontology_detail', 'license': {'label': 'CC-BY 4.0', 'logo': 'http://mirrors.creativecommons.org/presskit/buttons/80x15/png/by.png', 'url': 'https://creativecommons.org/licenses/by/4.0/'}, 'ontology_purl': 'http://purl.obolibrary.org/obo/chebi.owl', 'page': 'http://www.ebi.ac.uk/chebi/init.do?toolBarForward=userManual', 'products': [{'id': 'chebi.owl', 'ontology_purl': 'http://purl.obolibrary.org/obo/chebi.owl'}, {'id': 'chebi.obo', 'ontology_purl': 'http://purl.obolibrary.org/obo/chebi.obo'}, {'id': 'chebi.owl.gz', 'ontology_purl': 'http://purl.obolibrary.org/obo/chebi.owl.gz', 'title': 'chebi, compressed owl'}, {'id': 'chebi/chebi_lite.obo', 'ontology_purl': 'http://purl.obolibrary.org/obo/chebi/chebi_lite.obo', 'title': 'chebi_lite, no syns or xrefs'}, {'id': 'chebi/chebi_core.obo', 'ontology_purl': 'http://purl.obolibrary.org/obo/chebi/chebi_core.obo', 'title': 'chebi_core, no xrefs'}], 'title': 'Chemical Entities of Biological Interest', 'tracker': 'https://github.com/ebi-chebi/ChEBI/issues', 'twitter': 'chebit'}
# {'DO wiki': 'http://diseaseontology.sourceforge.net/', 'activity_status': 'active', 'browsers': [{'label': 'DO', 'title': 'DO Browser', 'url': 'http://www.disease-ontology.org/'}], 'build': {'infallible': 1, 'method': 'obo2owl', 'source_url': 'https://raw.githubusercontent.com/DiseaseOntology/HumanDiseaseOntology/master/src/ontology/doid.obo'}, 'contact': {'email': 'lynn.schriml@gmail.com', 'github': 'lschriml', 'label': 'Lynn Schriml'}, 'description': 'An ontology for describing the classification of human diseases organized by etiology.', 'domain': 'disease', 'facebook': 'https://www.facebook.com/diseaseontology', 'homepage': 'http://www.disease-ontology.org', 'id': 'doid', 'in_foundry_order': 1, 'layout': 'ontology_detail', 'license': {'label': 'CC0 1.0 Universal', 'logo': 'http://mirrors.creativecommons.org/presskit/buttons/80x15/png/cc-zero.png', 'url': 'https://creativecommons.org/publicdomain/zero/1.0/'}, 'ontology_purl': 'http://purl.obolibrary.org/obo/doid.owl', 'products': [{'id': 'doid.owl', 'ontology_purl': 'http://purl.obolibrary.org/obo/doid.owl', 'title': "Disease Ontology, OWL format. This file contains DO's is_a asserted hierarchy plus equivalent axioms to other OBO Foundry ontologies."}, {'id': 'doid.obo', 'ontology_purl': 'http://purl.obolibrary.org/obo/doid.obo', 'title': 'Disease Ontology, OBO format. This file omits the equivalent axioms.'}], 'publications': [{'id': 'http://www.ncbi.nlm.nih.gov/pubmed/?term=25348409', 'title': 'Disease Ontology 2015 update: an expanded and updated database of human diseases for linking biomedical knowledge through disease data'}], 'taxon': {'id': 'NCBITaxon:9606', 'label': 'Homo sapiens'}, 'title': 'Human Disease Ontology', 'tracker': 'https://github.com/DiseaseOntology/HumanDiseaseOntology/issues', 'twitter': 'diseaseontology', 'usages': [{'description': 'MGI disease model annotations use DO', 'examples': [{'description': 'Human genes and mouse homology associated with nail diseases', 'url': 'http://www.informatics.jax.org/disease/DOID:4123'}], 'user': 'http://www.informatics.jax.org/disease'}]}
# {'activity_status': 'active', 'browsers': [{'label': 'AmiGO', 'title': 'Gene Ontology AmiGO 2 Browser', 'url': 'http://amigo.geneontology.org/amigo/term/GO:0008150#display-lineage-tab'}], 'contact': {'email': 'suzia@stanford.edu', 'label': 'Suzi Aleksander'}, 'dependencies': [{'id': 'uberon', 'subset': 'go/extensions/uberon_import.owl'}, {'id': 'cl', 'subset': 'go/extensions/cl_import.owl'}, {'id': 'ncbitaxon', 'subset': 'go/extensions/ncbitaxon_import.owl'}, {'id': 'ro', 'subset': 'go/extensions/ro_import.owl'}, {'connects': [{'id': 'nifstd'}, {'id': 'go'}], 'description': 'Bridging axioms between nifstd and go', 'id': 'go/extensions/go-bridge-to-nifstd.owl', 'publications': [{'id': 'http://www.ncbi.nlm.nih.gov/pubmed/24093723', 'title': 'The Gene Ontology (GO) Cellular Component Ontology: integration with SAO (Subcellular Anatomy Ontology) and other recent developments.'}], 'title': 'GO bridge to NIFSTD', 'type': 'BridgeOntology'}], 'depicted_by': 'http://geneontology.org/assets/go-logo-icon.mini.png', 'description': 'An ontology for describing the function of genes and gene products', 'domain': 'biology', 'facebook': 'https://www.facebook.com/Gene-Ontology-305908656519/', 'homepage': 'http://geneontology.org/', 'id': 'go', 'in_foundry_order': 1, 'integration_server': 'http://build.berkeleybop.org/view/GO', 'label': 'GO', 'layout': 'ontology_detail', 'license': {'label': 'CC BY 4.0', 'logo': 'http://mirrors.creativecommons.org/presskit/buttons/80x15/png/by.png', 'url': 'https://creativecommons.org/licenses/by/4.0/'}, 'ontology_purl': 'http://purl.obolibrary.org/obo/go.owl', 'products': [{'description': 'The main ontology in OWL. This is self contained and does not have connections to other OBO ontologies', 'id': 'go.owl', 'ontology_purl': 'http://purl.obolibrary.org/obo/go.owl', 'page': 'http://geneontology.org/page/download-ontology', 'title': 'GO (OWL edition)'}, {'description': 'Equivalent to go.owl, in obo format', 'id': 'go.obo', 'ontology_purl': 'http://purl.obolibrary.org/obo/go.obo', 'page': 'http://geneontology.org/page/download-ontology', 'title': 'GO (OBO Format edition)'}, {'description': 'Equivalent to go.owl, in obograph json format', 'id': 'go.json', 'ontology_purl': 'http://purl.obolibrary.org/obo/go.json', 'page': 'https://github.com/geneontology/obographs/', 'title': 'GO (JSON edition)'}, {'description': 'The main ontology plus axioms connecting to select external ontologies', 'id': 'go/extensions/go-plus.owl', 'ontology_purl': 'http://purl.obolibrary.org/obo/go/extensions/go-plus.owl', 'page': 'http://geneontology.org/page/download-ontology', 'title': 'GO-Plus'}, {'description': 'As go-plus.owl, in obographs json format', 'id': 'go/extensions/go-plus.json', 'ontology_purl': 'http://purl.obolibrary.org/obo/go/extensions/go-plus.json', 'page': 'https://github.com/geneontology/obographs/', 'title': 'GO-Plus'}, {'description': 'The main ontology plus axioms connecting to select external ontologies', 'id': 'go/go-basic.obo', 'ontology_purl': 'http://purl.obolibrary.org/obo/go/go-basic.obo', 'page': 'http://geneontology.org/page/download-ontology', 'title': 'GO-Basic, Filtered, for use with legacy tools'}, {'description': 'As go-basic.obo, in json format', 'id': 'go/go-basic.json', 'ontology_purl': 'http://purl.obolibrary.org/obo/go/go-basic.json', 'page': 'http://geneontology.org/page/download-ontology', 'title': 'GO-Basic, Filtered, for use with legacy tools (JSON)'}, {'description': 'Classes added to ncbitaxon for groupings such as prokaryotes', 'id': 'go/extensions/go-taxon-groupings.owl', 'ontology_purl': 'http://purl.obolibrary.org/obo/go/extensions/go-taxon-groupings.owl', 'page': 'http://geneontology.org/page/download-ontology', 'title': 'GO Taxon Groupings'}, {'description': 'Equivalent to go.owl, but released daily. Note the snapshot release is not archived.', 'id': 'snapshot/go.owl', 'ontology_purl': 'http://purl.obolibrary.org/obo/snapshot/go.owl', 'page': 'http://geneontology.org/page/download-ontology', 'title': 'GO (OWL edition), daily snapshot release'}, {'description': 'Equivalent to go.owl, but released daily. Note the snapshot release is not archived.', 'id': 'snapshot/go.obo', 'ontology_purl': 'http://purl.obolibrary.org/obo/snapshot/go.obo', 'page': 'http://geneontology.org/page/download-ontology', 'title': 'GO (OBO Format edition), daily snapshot release'}], 'taxon': {'id': 'NCBITaxon:1', 'label': 'All life'}, 'termgenie': 'http://go.termgenie.org', 'title': 'Gene Ontology', 'tracker': 'https://github.com/geneontology/go-ontology/issues/', 'twitter': 'news4go', 'usages': [{'description': 'The GO ontology is used by the GO consortium for functional annotation of genes', 'examples': [{'description': 'annotations to transmembrane transport', 'url': 'http://amigo.geneontology.org/amigo/term/GO:0055085'}], 'type': 'annotation', 'user': 'http://geneontology.org'}]}

apikey="f48c1441acc3ac6146d19989f1cbb3de8608"
#
# e = entrezpy.esearch.esearcher.Esearcher("ontologysearch",
#                                          "leipzig@gmail.com",
#                                          apikey=apikey,
#                                          apikey_var=None,
#                                          threads=None,
#                                          qid=None)

with open('ontologies.jsonld') as json_file:
    onts = json.load(json_file)
    for f in onts['ontologies']:
        
        #bash = "./scholar.py -a '{}' --csv".format(f['title'])
        #print(bash)
        es = entrezpy.esearch.esearcher.Esearcher('esearcher', "leipzig@gmail.com",apikey=apikey,threads=1)
        a = es.inquire({'db':'pubmed','term':'({})'.format(f['title'].replace(" ", "+")), 'retmax': 110000, 'rettype': 'uilist'})
        #a = es.inquire('db':'nucleotide','term':'viruses[orgn]', 'retmax': 110000, 'rettype': 'uilist')
        b=a.get_result().uids
        print("{}\t{}".format(f['title'],len(b)))
        # analyzer = e.inquire({'db' : 'pubmed',
        #                       'id' : "{}".format(f['title']),
        #                       'retmode' : 'text',
        #                       'rettype': 'uilist'})
        #args = shlex.split(bash)
        #result = subprocess.run(args, stdout=subprocess.PIPE)
        #print(result.stdout)